injectorpp is flakey for Linux amd64

Here is the log https://github.com/microsoft/injectorppforrust/actions/runs/15263583121/job/42925345508?pr=12

Run if [[ "x86_64-unknown-linux-gnu" == "aarch64-unknown-linux-gnu" ]]; then
 Downloading crates ...
  Downloaded tokio-macros v2.5.0
  Downloaded unicode-ident v1.0.1[8](https://github.com/microsoft/injectorppforrust/actions/runs/15263583121/job/42925345508?pr=12#step:8:9)
  Downloaded quote v1.0.40
  Downloaded proc-macro2 v1.0.[9](https://github.com/microsoft/injectorppforrust/actions/runs/15263583121/job/42925345508?pr=12#step:8:10)5
  Downloaded pin-project-lite v0.2.16
  Downloaded syn v2.0.[10](https://github.com/microsoft/injectorppforrust/actions/runs/15263583121/job/42925345508?pr=12#step:8:11)1
  Downloaded tokio v1.45.1
   Compiling proc-macro2 v1.0.95
   Compiling unicode-ident v1.0.18
   Compiling libc v0.2.172
   Compiling quote v1.0.40
   Compiling pin-project-lite v0.2.16
   Compiling injectorpp v0.3.2 (/home/runner/work/injectorppforrust/injectorppforrust)
   Compiling syn v2.0.101
   Compiling tokio-macros v2.5.0
   Compiling tokio v1.45.1
    Finished `test` profile [unoptimized + debuginfo] target(s) in 7.17s
     Running unittests src/lib.rs (target/x86_64-unknown-linux-gnu/debug/deps/injectorpp-52ee56c81213dd1e)

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
     Running tests/async.rs (target/x86_64-unknown-linux-gnu/debug/deps/async-a8f7256e241051df)


running 3 tests
test test_complex_struct_async_func_with_param_should_success ... ok
test test_simple_async_func_should_success ... ok
test test_complex_struct_async_func_without_param_should_success ... ok

test result: ok. 3 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running tests/will_execute.rs (target/x86_64-unknown-linux-gnu/debug/deps/will_execute-2b52edc4aeff7495)

running 16 tests
test test_will_execute_when_fake_file_dependency_should_success ... ok
test test_will_execute_when_fake_file_dependency_should_success_times ... ok
test test_will_execute_when_fake_generic_function_multiple_types_should_success ... ok
test test_will_execute_when_fake_generic_function_single_type_can_recover ... ok
test test_will_execute_when_fake_generic_function_single_type_should_success ... ok

thread 'test_will_execute_when_fake_no_return_function_over_called_should_panic' panicked at tests/will_execute.rs:[11](https://github.com/microsoft/injectorppforrust/actions/runs/15263583121/job/42925345508?pr=12#step:8:12)8:23:
test test_will_execute_when_fake_method_can_recover ... ok
Fake function called more times than expected
test test_will_execute_when_fake_method_should_success ... ok
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

thread 'test_will_execute_when_fake_no_return_function_over_called_should_panic' panicked at src/interface/injector.rs:410:17:
Fake function was expected to be called 2 time(s), but it is actually called 3 time(s)

thread 'test_will_execute_when_fake_no_return_function_under_called_should_panic' panicked at src/interface/injector.rs:410:17:
Fake function was expected to be called 3 time(s), but it is actually called 2 time(s)
     Running tests/will_execute_raw.rs (target/x86_64-unknown-linux-gnu/debug/deps/will_execute_raw-75a926d51c40a0bb)
test test_will_execute_when_fake_method_with_output_parameter_no_return_should_success ... ok
test test_will_execute_when_fake_method_with_parameter_should_success ... ok
test test_will_execute_when_fake_multiple_reference_param_function_should_success ... ok
test test_will_execute_when_fake_multiple_reference_param_no_return_function_should_success ... ok
test test_will_execute_when_fake_no_return_function_over_called_should_panic - should panic ... ok
test test_will_execute_when_fake_no_return_function_should_success ... ok
test test_will_execute_when_fake_no_return_function_under_called_should_panic - should panic ... ok
test test_will_execute_when_fake_single_reference_param_function_should_success ... ok
test test_will_execute_when_fake_single_reference_param_no_return_function_should_success ... ok

test result: ok. 16 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s


running 6 tests
test test_will_execute_raw_when_fake_generic_function_multiple_types_should_success ... ok
fake_path_exists executed.
test test_will_execute_raw_when_fake_file_dependency_should_success ... ok
test test_will_execute_raw_when_fake_generic_function_multiple_types_with_different_conditins_should_success ... ok
test test_will_execute_raw_when_fake_generic_function_single_type_should_success ... ok
test test_will_execute_raw_when_fake_no_return_function_should_success ... ok
test test_will_execute_raw_when_fake_no_return_function_use_closure_should_success ... ok

test result: ok. 6 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running tests/will_return.rs (target/x86_64-unknown-linux-gnu/debug/deps/will_return-e3ec45ec934782d8)

running 4 tests
test test_will_return_boolean_when_fake_complex_generic_function_multiple_types_should_success ... ok

test test_will_return_boolean_when_fake_return_false_should_success ... ok
thread 'test_will_return_boolean_when_in_scope_should_restore' panicked at tests/will_return.rs:[13](https://github.com/microsoft/injectorppforrust/actions/runs/15263583121/job/42925345508?pr=12#step:8:14):5:
assertion `left == right` failed
  left: true
 right: false
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
test test_will_return_boolean_when_in_scope_should_restore ... FAILED
test test_will_return_boolean_when_fake_return_true_should_success ... ok

failures:

failures:
    test_will_return_boolean_when_in_scope_should_restore

test result: FAILED. 3 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

error: test failed, to rerun pass `--test will_return`
Error: Process completed with exit code 101.

--- a/src/interface/injector.rs
+++ b/src/interface/injector.rs
@@ -421,9 +421,9 @@ impl Drop for CallCountVerifier {
 /// A high-level type that holds patch guards so that when it goes out of scope,
 /// the original function code is automatically restored.
 pub struct InjectorPP {
+    _lock: MutexGuard<'static, ()>,
     guards: Vec<PatchGuard>,
     verifiers: Vec<CallCountVerifier>,
-    _lock: MutexGuard<'static, ()>,
 }
 
 impl InjectorPP {

--- a/tests/will_return.rs
+++ b/tests/will_return.rs
@@ -8,7 +8,7 @@ fn complex_generic_multiple_types_func_return_false<A, B, C>(_a: A, _b: B, _c: C
     return false;
 }
 
-/*#[test]
+#[test]
 fn test_will_return_boolean_when_in_scope_should_restore() {
     assert_eq!(returns_false(), false);
 
@@ -25,7 +25,7 @@ fn test_will_return_boolean_when_in_scope_should_restore() {
     let restored = returns_false();
 
     assert_eq!(restored, false);
-}*/
+}
 
 #[test]
 fn test_will_return_boolean_when_fake_return_true_should_success() {

